master/master.cfg

```diff
@@ -223,21 +223,22 @@ def get_delay(now, end):
 def no_builds_between(start, end):
     start = datetime.strptime(start, "%H:%M").time()
     end = datetime.strptime(end, "%H:%M").time()
-    def f(builder, requests):
+    @defer.inlineCallbacks
+    def canStartBuild(builder, wfb, request):
         now = datetime.now().time()
         if is_within_time_range(now, start, end):
             delay = get_delay(now, end)
-            # Schedule the build later
-            deferred = defer.Deferred()
-            builder.master.reactor.callLater(
-                int(delay),
-                deferred.callback,
-                requests[0],
```

Comment:
I think the problem is here, where the first element of requests is a "random" job that gets deferred.